src/lang.rs

```diff
@@ -431,12 +439,20 @@ named!(column_ref<Span, Expr>, do_parse!(
     (Expr::Column { head: DataAccessAtom::Key(head), rest: rest })
 ));
 
-named!(ident<Span, String>, do_parse!(
+named!(ident<Span, String>, alt!(bare_ident | escaped_ident));
+
+named!(bare_ident<Span, String>, do_parse!(
     start: take_while1!(starts_ident) >>
     rest: take_while!(is_ident) >>
     (start.fragment.0.to_owned() + rest.fragment.0)
 ));
 
+named!(escaped_ident<Span, String>, do_parse!(
```

Comment:
what if you use the `quoted_string` construction instead? that will handle considerably more flexible characters (including escaping quotes inside of the string)

Reply:
Good call!  I've swapped in `quoted_string` as you suggested and added a new test case in `parse_quoted_ident`.

Technically I think JSON and [logfmt](https://discuss.python.org/t/logfmt-compatible-mode-for-debugging-specifier/7304) require double quotes for strings, but I think it's probably better to be permissive since we can receive unstructured input.